pkg/auth/dependency/sso/google.go

```diff
@@ -1,58 +1,73 @@
 package sso
 
-import "github.com/skygeario/skygear-server/pkg/core/skyerr"
+import (
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+const (
+	googleAuthorizationURL string = "https://accounts.google.com/o/oauth2/v2/auth"
+	// nolint: gosec
+	googleTokenURL    string = "https://www.googleapis.com/oauth2/v4/token"
+	googleUserInfoURL string = "https://www.googleapis.com/oauth2/v1/userinfo"
+)
 
 type GoogleImpl struct {
-	Setting Setting
-	Config  Config
+	OAuthConfig    config.OAuthConfiguration
+	ProviderConfig config.OAuthProviderConfiguration
 }
 
 func (f *GoogleImpl) GetAuthURL(params GetURLParams) (string, error) {
-	if f.Config.ClientID == "" {
-		skyErr := skyerr.NewError(skyerr.InvalidArgument, "ClientID is required")
-		return "", skyErr
-	}
 	params.Options["access_type"] = "offline"
```

Comment:
I think we should use `online` at the moment; we do not need refresh tokens.

Reply:
I didn't change that part of code. Though I agree we can use `online`.